packages/camera/camera_android_camerax/test/rotated_preview_test.dart

```diff
@@ -0,0 +1,150 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:camera_android_camerax/src/rotated_preview.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  double getRotationDegreesFromDeviceOrientation(
+      DeviceOrientation deviceOrientation) {
+    return switch (deviceOrientation) {
+      DeviceOrientation.portraitUp => 0,
+      DeviceOrientation.landscapeRight => 90,
+      DeviceOrientation.portraitDown => 180,
+      DeviceOrientation.landscapeLeft => 270,
+    };
+  }
+
+  double calculateExpectedPreviewRotation(double sensorOrientationDegrees,
+      double deviceOrientationDegrees, int facingSign) {
+    return ((sensorOrientationDegrees -
+                deviceOrientationDegrees * facingSign +
+                360) %
+            360) -
+        deviceOrientationDegrees;
+  }
+
+  void findAndTestRotatedPreviewHasExpectedRotation(
```

Comment:
I already left comments about changing this test, but I wanted to comment on this.

This is a duplication of the implementation code, so we now we have the possibility of getting it wrong, twice, but all the tests passing.

Instead, like I mentioned in the previous file, I'd want to see something like:

"When the camera preview does not handle crop and rotation, and sensor is X, and device is Y, the preview is rotated 90 degrees clockwise"

In fact, the test case hopefully would look like that:

```dart
const _$90DegreesClockwise = 1;

RotatedBox widget = ...
expect(
  widget.quarterTurns,
  _$90DegreesClockwise,
  reason: 'Expected 1 quarter turn, which is 90 degrees clockwise',
);
```

It will help a lot with making the error messages (on CI and locally) readable as well.